src/storage/secondary/block/block_index_builder.rs

```diff
@@ -42,14 +42,15 @@ impl BlockIndexBuilder {
         column_data: &mut Vec<u8>,
         block_data: &mut Vec<u8>,
         stats: Vec<BlockStatistics>,
+        first_key: &mut Vec<u8>,
     ) {
         self.indexes.push(BlockIndex {
             offset: column_data.len() as u64,
             length: block_data.len() as u64 + BLOCK_HEADER_SIZE as u64,
             first_rowid: self.last_row_count as u32,
             row_count: (self.row_count - self.last_row_count) as u32,
             /// TODO(chi): support sort key
-            first_key: "".into(),
+            first_key: first_key.drain(..).collect(),
```

Comment:
I'd prefer to get a full ownership of `first_key` in this function. Otherwise this interface would look confusing 🤣